storage/local/crashrecovery.go

```diff
@@ -60,11 +60,12 @@ func (p *persistence) recoverFromCrash(fingerprintToSeries map[model.Fingerprint
 			continue
 		}
 		if err != nil {
+			dir.Close()
```

Comment:
In case of an error, there are no guarantees about the state of `dir`, it is most likely `nil`. Close should only be called if no error was returned, so remove this line to restore the previous behavior.